### PR TITLE
feat(arm): ACREnableZoneRedundancy

### PR DESCRIPTION
**Pull request by channy-katz**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
*adds a ACREnableZoneRedundancy*

### Description
*The function checks whether the "zoneRedundancy" property is set to True for a resource and all its replications. If any "zoneRedundancy" is not True, it returns a failure result; otherwise, it returns a pass result.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/ACREnableZoneRedundancy.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        categories = (CheckCategories.BACKUP_AND_RECOVERY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        # check registry. default=false
+        if conf.get("properties").get("zoneRedundancy", []) != [True]:
+            return CheckResult.FAILED
+
+        # check each replica. default=false
+        replications = conf.get("replications", {})
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/ACREnableZoneRedundancy.py:28]: I don't see the `replications` field in `Microsoft.ContainerRegistry/registries/replications` or `Microsoft.ContainerRegistry/registries`, are you sure this is the field we should check here?

Comment (channy-katz) [reply checkov/arm/checks/resource/ACREnableZoneRedundancy.py:28]: I changed the error
can you confirm me please

Comment (ChanochShayner) [reply checkov/arm/checks/resource/ACREnableZoneRedundancy.py:28]: My comment was for the resource template, Are you sure that those resources have the field `replications` to configure?

Comment (channy-katz) [reply checkov/arm/checks/resource/ACREnableZoneRedundancy.py:28]: I checked and I think it's good now

Comment (ChanochShayner) [reply checkov/arm/checks/resource/ACREnableZoneRedundancy.py:28]: You are looking for the field `replications` in the `Microsoft.ContainerRegistry/registries` resource type
but I don't see [here](https://learn.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries?pivots=deployment-language-arm-template) the `replications` filed

**Comment by channy-katz** [comment]
I changed the mistakes 
But there was something about the title, I don't find a problem there

File: checkov/arm/checks/resource/ACREnableZoneRedundancy.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        id = "CKV_AZURE_233"
+        supported_resources = ("Microsoft.ContainerRegistry/registries", "Microsoft.ContainerRegistry/registries/replications",)
+        categories = (CheckCategories.BACKUP_AND_RECOVERY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        # check registry. default=false
+        properties = conf.get("properties")
+        if properties and isinstance(properties, dict):
+            if properties.get("zoneRedundancy", []) != [True]:
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/ACREnableZoneRedundancy.py:26]: zoneRedundancy should be Enabled or Disabled and not a boolean value

**Review by ChanochShayner — COMMENTED** [review]
your UT is failing, please fix it and the other failing jobs.

File: checkov/arm/checks/resource/ACREnableZoneRedundancy.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        id = "CKV_AZURE_233"
+        supported_resources = ("Microsoft.ContainerRegistry/registries", "Microsoft.ContainerRegistry/registries/replications",)
+        categories = (CheckCategories.BACKUP_AND_RECOVERY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        # check registry. default=false
+        properties = conf.get("properties")
+        if properties and isinstance(properties, dict):
+            if properties.get("zoneRedundancy") == "disabled":
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/ACREnableZoneRedundancy.py:26]:
```suggestion
            if properties.get("zoneRedundancy") == "Disabled":
```

File: tests/arm/checks/resource/example_ACREnableZoneRedundancy/fail1.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        "tier": "[parameters('acrSku')]"
+      },
+      "tags": {
+        "displayName": "Container Registry",
+        "container.registry": "[parameters('acrName')]"
+      },
+      "properties": {
+        "adminUserEnabled": "[parameters('acrAdminUserEnabled')]",
+        "zoneRedundancy":
+          "disabled"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_ACREnableZoneRedundancy/fail1.json:22]:
```suggestion
          "Disabled"
```

File: tests/arm/checks/resource/example_ACREnableZoneRedundancy/fail2.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    {
+      "type": "Microsoft.ContainerRegistry/registries/replications",
+      "apiVersion": "2020-11-01",
+      "name": "fail2",
+      "location": "[parameters('acrReplicaLocation')]",
+      "dependsOn": [
+        "[resourceId('Microsoft.ContainerRegistry/registries/', parameters('acrName'))]"
+      ],
+      "properties": {
+        "zoneRedundancy": "disabled"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_ACREnableZoneRedundancy/fail2.json:14]:
```suggestion
        "zoneRedundancy": "Disabled"
```

File: tests/arm/checks/resource/example_ACREnableZoneRedundancy/pass.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+          "name": "[parameters('acrSku')]",
+          "tier": "[parameters('acrSku')]"
+        },
+        "tags": {
+          "displayName": "Container Registry",
+          "container.registry": "[parameters('acrName')]"
+        },
+        "properties": {
+          "adminUserEnabled": "[parameters('acrAdminUserEnabled')]",
+          "zoneRedundancy": "enabled"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_ACREnableZoneRedundancy/pass.json:21]:
```suggestion
          "zoneRedundancy": "Enabled"
```

File: tests/arm/checks/resource/example_ACREnableZoneRedundancy/pass2.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  "resources": [{
+  "type": "Microsoft.ContainerRegistry/registries/replications",
+  "apiVersion": "2020-11-01",
+  "name": "pass2",
+  "location": "[parameters('acrReplicaLocation')]",
+  "dependsOn": [
+    "[resourceId('Microsoft.ContainerRegistry/registries/', parameters('acrName'))]"
+  ],
+  "properties": {
+    "zoneRedundancy": "enabled"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_ACREnableZoneRedundancy/pass2.json:13]:
```suggestion
    "zoneRedundancy": "Enabled"
```

**Review by ChanochShayner — APPROVED** [review]

**Review by achiar99 — APPROVED** [review]

**Pull request merged**